signals.go

```diff
@@ -18,14 +18,23 @@ const signalBufferSize = 2048
 // while still forwarding all other signals to the process.
 // If notifySocket is present, use it to read systemd notifications from the container and
 // forward them to notifySocketHost.
-func newSignalHandler(enableSubreaper bool, notifySocket *notifySocket) chan *signalHandler {
+func newSignalHandler(enableSubreaper bool, detach bool, notifySocket *notifySocket) chan *signalHandler {
 	if enableSubreaper {
 		// set us as the subreaper before registering the signal handler for the container
 		if err := system.SetSubreaper(1); err != nil {
 			logrus.Warn(err)
 		}
 	}
-	handler := make(chan *signalHandler)
+	handler := make(chan *signalHandler, 1)
```

Comment:
Okay, so the channel is buffered now. Do we expect to send only 1 msg?

Reply:
I think so, yes (although I don't know why @lifubang made it buffered).

Reply:
> Do we expect to send only 1 msg?

Yes

> although I don't know why @lifubang made it buffered

Because I don’t want to open a new go routine for detached containers.
Welcome other solution suggestions which don’t use a buffered channel.

Reply:
After moved notifySocket out of signalHandler now, we can change it back now.